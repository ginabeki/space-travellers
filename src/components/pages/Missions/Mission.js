/* eslint-disable react/prop-types */
import React from 'react';
import { Button } from 'react-bootstrap';
import Badge from 'react-bootstrap/Badge';
import { useDispatch } from 'react-redux';
import Table from 'react-bootstrap/Table';
import { joinMission, leaveMission } from '../../../redux/missions/missions';

export default function Mission(props) {
  const { missions } = props;
  const dispatch = useDispatch();
  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Missions</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions
         && missions.map((mission) => (
           <tr key={mission.id}>
             <td>{mission.missionName}</td>
             <td>{mission.description}</td>
             <td>
               {!mission.reserved
                 ? <Badge bg="secondary">Not A Member</Badge>
                 : <Badge bg="primary">Active Member</Badge>}
             </td>
             <td>
               {mission.reserved
                 ? <Button variant="outline-danger" onClick={() => dispatch(leaveMission(mission.id))}>Leave Mission</Button>
                 : <Button variant="outline-secondary" onClick={() => dispatch(joinMission(mission.id))}>Join Mission</Button>}
             </td>
           </tr>
         ))}

      </tbody>
    </Table>
  );
}
