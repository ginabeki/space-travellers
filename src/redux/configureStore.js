import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { missionsReducer } from './missions/missions';
import { rocketsReducer } from './rockets/rockets';

const rootReducer = combineReducers({
  missions: missionsReducer,
  rockets: rocketsReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});

export default { store };
