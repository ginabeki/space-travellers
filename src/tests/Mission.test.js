import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/configureStore';
import Mission from '../components/pages/Missions/Mission';

describe('Mission Component Test ', () => {
  it('Expected to render Mission contents', () => {
    const ProfileComponent = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Mission />
        </Router>
      </Provider>,
    ).toJSON();
    expect(ProfileComponent).toMatchSnapshot();
  });
});
