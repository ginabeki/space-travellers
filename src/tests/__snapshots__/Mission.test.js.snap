// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component Test  Expected to render MyProfile contents 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Missions
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
